main.go

```diff
@@ -50,7 +50,12 @@ func _main() int {
 		return 1
 	}
 
-	discoverController := k8sapi.NewController(kubeClient)
+	discoverController, err := k8sapi.NewController(kubeClient)
+	if err != nil {
+		log.Errorf("Failed to create new Discover Controller: %v", err)
+		return 1
+	}
+
 	go discoverController.DiscoverK8SPods()
```

Comment:
This needs to go inside the `NewController()` function and be synchronously executed (see below)